packages/block-library/src/navigation-menu-item/save.js

```diff
@@ -1,12 +1,32 @@
-export default function save( { attributes } ) {
+/**
+ * WordPress dependencies
+ */
+import {
+	InnerBlocks,
+} from '@wordpress/block-editor';
+
+export default function save( { attributes/*, innerBlocks*/ } ) {
+	const { destination, nofollow, title, opensInNewTab, label } = attributes;
 	return (
-		<a
-			href={ attributes.destination }
-			rel={ attributes.nofollow && 'nofollow' }
-			title={ attributes.title }
-			target={ attributes.opensInNewTab && '_blank' }
-		>
-			{ attributes.label }
-		</a>
+		<li>
+			<a
+				href={ destination }
+				rel={ nofollow ? 'nofollow' : undefined }
+				title={ title }
+				target={ opensInNewTab ? '_blank' : undefined }
+				className="wp-block-navigation-menu-item"
+			>
+				{ label }
+			</a>
+			<ul>
+				<InnerBlocks.Content />
+			</ul>
+
+			{ /* This should make the HTML look the same but it invalidates the block */ }
+
+			{ /* ( innerBlocks.length > 0 ) && <ul>
+				<InnerBlocks.Content />
+			</ul> */ }
```

Comment:
@noisysocks after @jorgefilipecosta  pointed out to me that innerBlocks is passed to the save function I thought that it would be as easy as in the commented code above. Unfortunately that code gets the block invalidated as soon as I reload load the page ... the markup in code view is good, but somehow the post still gets saved with empty UL s for childless items.

Reply:
Also there is the option to drop any saved content and make the navigation-menu-item block dynamic as well. That @noisysocks would "solve" all of our problems :)